content/contribute/contributing-and-managing-patterns.adoc

```diff
@@ -0,0 +1,145 @@
+----
+
+If you want to modify and track your version of common, fork and clone the common repository separately:
+
+
+[source,terminal]
+----
+$ git clone https://github.com/<your-workspace>/common
+----
+
+Make changes in your fork’s main branch or create a new branch for your modifications.
+
+To track these changes in your fork of the pattern repository (e.g., `industrial-edge`), replace the common subtree with your forked version. To simplify this process use:
+
+[source,terminal]
+----
+$ common/scripts/make_common_subtree.sh <subtree_repo> <subtree_branch> <subtree_remote_name>
+----
+
+This script sets up a new remote in your local working directory with the specified repository. It replaces the common directory with the specified fork and branch, commits the change, but does not push it.
+
+For example:
+
+[source,terminal]
+----
+$ common/scripts/make_common_subtree.sh https://github.com/mhjacks/common.git wip-main common-subtree
+----
+
+This replaces the common directory in the current repository with the wip-main branch from mhjacks's common repository and names the remote `common-subtree`.
+
+To pull changes from mhjacks's wip-main branch into the parent repository:
+
+[source,terminal]
+----
+$ git subtree pull --prefix common common-subtree wip-main
+----
+
+Running the script without arguments defaults to:
+
+[source,terminal]
+----
+$ common/scripts/make_common_subtree.sh https://github.com/validatedpatterns/common.git main common-subtree
+----
+
+These are the default settings for the repository configuration.
+
+[id="subtree-vs-submodule"]
+== Subtree vs. Submodule
+
+Ensuring patterns can be easily shared across multiple projects has always been a priority. While it is technically possible to share changes between unrelated Git repositories, this is a largely manual process prone to errors. We aim to offer a seamless "pull" experience, allowing a single `git pull` action to update shared pattern components. For repository sharing, we considered two main strategies: submodules and subtrees. Initially, we used submodules but later moved to subtrees.
+
+Subtrees integrate the history of another repository into a parent repository, offering many benefits of submodules without most of their drawbacks. Atlassian provides useful documentation on subtrees, which explains their advantages in more detail. For more information see, link:https://www.atlassian.com/blog/developer[Developer's blog] and link:https://www.atlassian.com/git/tutorials/git-subtree[Git subtree]. 
+
+Submodules presented unavoidable challenges, such as:
+
+* Remembering to check out repositories with `--recurse-submodules` or running `git submodule init && git submodule sync`. The common directory often appeared empty without these steps.
+* Ensuring compatibility between submodule-based repositories and other tools. While tools like ArgoCD and Tekton Pipelines supported submodules well, compatibility concerns remained.
```

Comment:
Do we really need the history lesson here - should we not simply state what method we use now, just my 2 cents